lib/OpenQA/Schema/Result/JobGroups.pm

```diff
@@ -166,12 +166,25 @@ sub _find_expired_jobs {
 
     # all jobs not in important builds that are expired
     my $timecond = {'<' => time2str('%Y-%m-%d %H:%M:%S', time - 24 * 3600 * $keep_in_days, 'UTC')};
-    push(@ors, {BUILD => {-not_in => $important_builds}, t_finished => $timecond});
+
+    # filter out linked jobs
+    my $expired_jobs = $self->jobs->search({BUILD => {-not_in => $important_builds}, t_finished => $timecond}, {order_by => qw(id)});
+    my @linked_jobs = ();
+    while (my $job = $expired_jobs->next) {
+        my $comments = $job->comments;
+        while (my $comment = $comments->next) {
+            if ($comment->label =~ /^linked/) {
```

Comment:
if someone made the effort to label it, she did more than just following a link. So yes, I consider that job to be more important than the rest.